test/framework/deployments/otelcollector/deployment_kubernetes.go

```diff
@@ -0,0 +1,322 @@
+			Labels: map[string]string{
+				"app": k.Name(),
+			},
+		},
+		Spec: corev1.PodSpec{
+			ServiceAccountName: k.serviceAccountName,
+			Containers: []corev1.Container{
+				{
+					Name:            k.Name(),
+					ImagePullPolicy: "IfNotPresent",
+					Image:           k.image,
+					Args:            []string{"--config=/conf/otel-collector-config.yaml"},
+					Resources: corev1.ResourceRequirements{
+						Limits: corev1.ResourceList{
+							"cpu":    resource.MustParse("500m"),
+							"memory": resource.MustParse("500Mi"),
+						},
+					},
+					Ports: []corev1.ContainerPort{
+						{
+							Name:          "grpc-otel",
+							ContainerPort: GRPCPort,
+						},
+						{
+							Name:          "http-otel",
+							ContainerPort: HTTPPort,
+						},
+						{
+							Name:          "metrics",
+							ContainerPort: MetricsPort,
+						},
+						{
+							Name:          "prometheus",
+							ContainerPort: PrometheusExporterPort,
+						},
+					},
+					Env: []corev1.EnvVar{
+						{Name: "MY_POD_IP", ValueFrom: &corev1.EnvVarSource{
+							FieldRef: &corev1.ObjectFieldSelector{
+								APIVersion: "v1",
+								FieldPath:  "status.podIP",
+							},
+						}},
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "otel-collector-config-vol",
+							ReadOnly:  true,
+							MountPath: "/conf",
+						},
+					},
+				},
+			},
+			Volumes: []corev1.Volume{
+				{
+					Name: "otel-collector-config-vol",
+					VolumeSource: corev1.VolumeSource{
+						ConfigMap: &corev1.ConfigMapVolumeSource{
+							LocalObjectReference: corev1.LocalObjectReference{Name: "otel-collector-conf"},
+							DefaultMode:          pointer.To(int32(0o555)),
+							Items: []corev1.KeyToPath{{
+								Key:  "config",
+								Path: "otel-collector-config.yaml",
+							}},
+						},
+					},
+				},
+			},
+		},
+	}
+	return spec
+}
+
+func (k *K8SDeployment) configMap() *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: meta(k.namespace, "otel-collector-conf", map[string]string{"app": k.Name()}),
+		Data: map[string]string{
+			"config": config(k.endpointBasedOnIP()),
+		},
+	}
+}
+
+func (k *K8SDeployment) serviceAccount() *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ServiceAccount",
+			APIVersion: "v1",
+		},
+		ObjectMeta: meta(k.namespace, k.serviceAccountName, map[string]string{"app": k.Name()}),
+	}
+}
+
+func meta(namespace string, name string, labels map[string]string) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:      name,
+		Namespace: namespace,
+		Labels:    labels,
+	}
+}
+
+func (k *K8SDeployment) endpointBasedOnIP() string {
+	if k.isIPv6 {
+		return "[${env:MY_POD_IP}]"
+	}
+	return "${env:MY_POD_IP}"
+}
+
+func config(endpoint string) string {
+	return fmt.Sprintf(`receivers:
+  otlp:
+    protocols:
+      grpc:
+        endpoint: "%s:4317"
```

Comment:
should this use these https://github.com/kumahq/kuma/pull/8893/files?w=1#diff-f37fdaffd729716e2d2f5c631df7103b5f13395f9a424127e23b1e1ad1ff881bR11-R14 ? to avoid confusion when changing one?

Reply:
changed